src/sparsezoo/refactor/directory.py

```diff
@@ -0,0 +1,194 @@
+# limitations under the License.
+
+import logging
+import os
+import pathlib
+import tarfile
+from typing import List, Optional
+
+from src.sparsezoo.refactor.file import File
+from src.sparsezoo.utils.downloader import download_file
+
+
+__all__ = ["Directory"]
+
+
+class Directory(File):
+    """
+    Object that represents a directory.
+    Directory may also represent a tar file.
+
+    :param files: list of files contained within the Directory
+        (None for tar files).
+    :param name: name of the Directory
+    :param path: path of the Directory
+    :param url: url of the Directory
+    """
+
+    def __init__(
+        self,
+        name: str,
+        files: Optional[List[File]] = None,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+
+        self.files = files
+
+        super().__init__(name=name, path=path, url=url)
+
+    def validate(self, strict_mode: bool = True):
+        """
+        Validates the structure of directory.
+
+        :return: boolean flag; True if files are valid and no errors arise
+        """
+        validations = {}
+        if self._is_tar():
+            # we are not validating tar files for now
+            validations[self.name] = True
+
+        else:
+            validations = {}
+            for file in self.files:
+                if isinstance(file, Directory):
+                    # we are not validating nested directories for now
+                    raise NotImplementedError(
+                        "Method validate() does not support nested directories yet."
+                    )
+                else:
+                    validations[file.name] = file.validate(strict_mode=strict_mode)
+
+        if not all(validations.values()):
+            logging.warning(
+                "Following files: "
+                f"{[key for key, value in validations.items() if not value]} "
+                "were not successfully validated."
+            )
+
+        return all(validations.values())
+
+    def download(self, destination_path: str, overwrite: bool = True):
+        """
+        Download the contents of the file from the url.
+
+        :param destination_path: the local file path to save the downloaded file to
+        :param overwrite: True to overwrite any previous files if they exist,
+            False to not overwrite and raise an error if a file exists
+        """
+        # Directory can represent a tar file.
+        if self._is_tar():
+            download_file(
+                url_path=self.url,
+                dest_path=os.path.join(destination_path, self.name),
+                overwrite=overwrite,
+            )
+            self.path = os.path.join(destination_path, self.name)
+
+        # Directory can represent a folder.
+        else:
+            for file in self.files:
+                if isinstance(file, Directory):
+                    raise NotImplementedError(
```

Comment:
any reason we can't recurse?

Reply:
We can and we should. Overlooked this, fix in progress.